GiniVision/Classes/Networking/AccountingDocumentService.swift

```diff
@@ -77,70 +87,52 @@ final class AccountingDocumentService: DocumentServiceProtocol {
                         fileName: String,
-                        docType: String = "",
-                        cancellationToken: BFCancellationToken? = nil,
+                        docType: Document.DocType? = nil,
                         completion: @escaping UploadDocumentCompletion) {
         Log(message: "Creating document...", event: "📝")
         
-        giniSDK.sessionManager
-            .getSession()
-            .continueWith(block: getSession(with: cancellationToken))
-            .continueOnSuccessWith(block: { [weak self] _ in
-                return self?.giniSDK.documentTaskManager.createDocument(withFilename: fileName,
-                                                                        from: document.data,
-                                                                        docType: docType,
-                                                                        metadata: self?.metadata,
-                                                                        cancellationToken: cancellationToken)
-            }).continueWith(block: { [weak self] task in
-                guard let self = self else { return nil }
-                if let createdDocument = task.result as? GINIDocument {
-                    Log(message: "Created document with id: \(createdDocument.documentId ?? "") " +
-                        "for vision document \(document.id)", event: "📄")
-                    
-                    self.document = createdDocument
-                    completion(.success(createdDocument))
-                } else if task.isCancelled {
-                    Log(message: "Document creation was cancelled", event: .error)
-                    completion(.failure(AnalysisError.cancelled))
-                } else {
-                    Log(message: "Document creation failed", event: .error)
-                    completion(.failure(AnalysisError.documentCreation))
-                }
-                
-                return nil
-            })
+        documentService.createDocument(with: document.data,
+                                       fileName: fileName,
+                                       docType: docType,
+                                       metadata: metadata) { result in
+                                        switch result {
+                                        case .success(let createdDocument):
+                                            Log(message: "Created document with id: \(createdDocument.id) " +
+                                                "for vision document \(document.id)", event: "📄")
+                                            
+                                            self.document = createdDocument
+                                            completion(.success(createdDocument))
+                                        case .failure(let error):
+                                            Log(message: "Document creation failed", event: .error)
```

Comment:
Here, you use a `Log` and other places you use `print`. I assume everything should be `Log`?

Reply:
No. I only use _Log_ within the library. The _print_s are part of the Example app.